lib/scope.js

```diff
@@ -2149,68 +2242,63 @@ module.exports = {
-      if ( from_story_table ) {
+      if ( from_story_table && !row.artificial ) {
         if ( row.var.match( /\.there_is_another$/ ) ) {
-          // `.there_is_another` should never get in here, but just in case
           await scope.addToReport( scope, { type: `warning`, value: `The attribute \`.there_is_another\` is invalid in story table tests. Replace it with \`.target_number\` in your \`var\` column. Set the \`value\` to the number of items in that list. This test will now set this row's \`value\` to \`False\`. See https://suffolklitlab.github.io/docassemble-AssemblyLine-documentation/docs/automated_integrated_testing#there_is_another-loop. The row data is\n${ JSON.stringify( row.original )}` });
         } else {
           // If `target_number` is set to True instead of a number
-          // Since secret vars can't be passed in a table, this should never be a secret var
           await scope.addToReport( scope, { type: `warning`, value: `${ val_to_print } value is not a valid value for ${ row.var } here. This test will default to \`False\` to avoid problems.` });
         }
         return `False`;
       } else { return `True`; }
-
-    // Story table stuff
-    // Target: 3 (3 items. press 'yes' twice.)
-    // Times used: 0 -> `True`, 1 -> `True`, 2 -> `False`, 
-    } else if ( row.times_used >= int_val - 1 ) {  // Story table loop will end
-        return `False`;
-
-    } else if ( row.source && row.source.times_used >= int_val - 1 ) {  // Just in case use changes a tad
-        return `False`;
-
-    } else if ( isNaN( parseInt( value )) ) { // Loop will end
-      // I can't figure out how to trigger this warning either
+    
+    } else if ( isNaN( desired_num )) {  // Story table loop will end
+      // I can't figure out how to trigger this warning, so maybe we're
+      // already handling everything we need to
       await scope.addToReport( scope, { type: `warning`, value: `${ val_to_print } value is not a valid value for ${ row.var } here. This test will default to \`False\` to avoid problems.` });
       return `False`;
 
-    } else {  // It's a number. Loop will continue.
+    // Story table stuff
+    // target_number: 3 (3 items. press 'yes' on there_is_another 2 times.)
+    // used_for[ <var_name>.target_number ]: 0 -> `True`, 1 -> `True`, 2 -> `False`
+
+    // Why `desired_num - 1`? The first `there_is_another` is actually the second loop
+    // (after they've already filled in the first list item info). If the user always had
+    // to set `there_are_any`, this wouldn't be a problem, but sometimes the author sets
+    // `there_are_any` beforehand instead so we can't depend on this, so we only count the
+    // use of `there_is_another` and make sure to count the 1st appearance as the second
+    // loop. (or the 0'th appearance as the first loop)
+    } else if ( row.source && row.source.used_for[ row.var ] >= ( desired_num - 1 )) {  // Story table loop will end
```

Comment:
After thinking about this, and seeing that you've written a lot trying to explain why we're doing `desired_num - 1`, I think it makes more sense to instead of `used_for + 1`; the number of times that you've already said "there is another" plus the first item is how many items are in the list.

```suggestion
    } else if ( row.source && row.source.used_for[ row.var ] + 1 >= desired_num) {  // Story table loop will end
```

Reply:
I like the more concise description! I do think it still needs some kind of explanation in a comment, though.

Reply:
Is the comments something you'd like to work on or should I give it a shot? Or is there a compelling reason to leave out comments there?

Reply:
Idk, I left it out because any comment here would just be describing the "what" of the code, which isn't going to
be that useful. I don't think we needed the part about trying to use `.there_are_any`; it was distracting and not relevant to what we are currently doing now.

```suggestion
    // check that the number of elements in the list (i.e. the time's we've asked for `.there_is_another`,
   // plus 1 for the first item before asking) hits our desired number. 
    } else if ( row.source && row.source.used_for[ row.var ] + 1 >= desired_num) {  // Story table loop will end
```

Reply:
How about the below?

```suggestion
	    // Check that the number of elements in the list hits our desired number.
	    // We have to use `+ 1` to represent that the first loop is handled
	    // by the value of the `.there_are_any` interview attribute instead.
	    } else if ( row.source && row.source.used_for[ row.var ] + 1 >= desired_num) {  // Story table loop will end
```